Jenkinsfile

```diff
@@ -716,6 +683,68 @@ stage('Non-default environments') {
 	}
 }
 
+stage('Sonar analysis') {
+	runBuildOnNode {
+		withMavenWorkspace {
+			dir(helper.configuration.maven.localRepositoryPath) {
+				unstash name: "default-build-cache"
+			}
+
+			unstash name: "default-build-jacoco-reports"
+			environments.content.jdk.enabled.each { JdkBuildEnvironment buildEnv ->
+				unstash name: "${buildEnv.tag}-build-jacoco-reports"
+			}
+			environments.content.database.enabled.each { JdkBuildEnvironment buildEnv ->
+				unstash name: "${buildEnv.tag}-build-jacoco-reports"
+			}
+			environments.content.localElasticsearch.enabled.each { JdkBuildEnvironment buildEnv ->
+				unstash name: "${buildEnv.tag}-build-jacoco-reports"
+			}
+			environments.content.amazonElasticsearch.enabled.each { JdkBuildEnvironment buildEnv ->
+				unstash name: "${buildEnv.tag}-build-jacoco-reports"
+			}
+
+			// we don't clean to keep the unstashed jacoco reports:
+			sh "mvn package -Pskip-checks -Pci-build -DskipTests -Pcoverage-report ${toTestEnvironmentArgs(environments.content.jdk.default)}"
+
+			def sonarCredentialsId = helper.configuration.file?.sonar?.credentials
+			if (sonarCredentialsId) {
```

Comment:
You can actually move this up to skip the node allocation (`runBuildOnNode`) when there are no sonar credentials... I think?